pythonPackages.PyLD: init at 0.7.2

apeschar opened the pull request:
###### Motivation for this change

Add PyLD python package.

Tested with Python 2, 3 and PyPy.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

apeschar committed:
pythonPackages.PyLD: init at 0.7.2

FRidh commented:
- see the header of `python-packages.nix` and the Nixpkgs manual about where your expression needs to be
- use the template that is provided when opening a PR

apeschar commented:
OK. Please see the updated text and commit.

FRidh commented:
Thanks. I see that you tested without sandboxing so that way I know that we should definitely test the expression as dependencies might be missing.

When building it, it shows 
```
Ran 0 tests in 0.000s
```
This is not good. We like to have tests, and it no tests are found then it either means the test runner is looking in the wrong place, or there simply are no tests. In the former case we fix the `checkPhase` to make it find the tests, and in the latter case we explicitly disable the `checkPhase` and leave a comment stating why the `checkPhase` is disabled.

apeschar commented:
Alright, I'm learning.

First of all, I now tested _with_ sandboxing. Works. (On all three interpreters.)

Regarding the tests:

The testcases are not included in the release archive, so we need to pull from GitHub, and also get two additional repos that contain the actual test data. (It's a official corpus with example inputs and outputs.)

When that is done, it turns out that the library fails a lot of test cases. In order to have some test coverage without having to dive into the library itself, I've set a minimum for the count of passed tests. That's something, at least.

What do you think?

FRidh commented:
It's great you've looked into this. Running the tests is in the end for you, as maintainer, to know that your package will still work. You can choose to disable them if you want, but there's the risk the package won't work. You can enable them, and that might take some work, as you've know noticed. Its up to you how far you want to go.

FRidh commented on pkgs/development/python-modules/PyLD/default.nix:
```diff
@@ -0,0 +1,58 @@
+    sha256 = "1bmpz4s6j7by6l45wwxy7dn7hmrhxc26kbx2hbfy41x29vbjg6j9";
+  };
+
+  inherit json-ld normalization;
```
no need for this

apeschar replied:
Missed that.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -31751,6 +31751,8 @@ EOF
 
   distro = callPackage ../development/python-modules/distro { };
 
+  PyLD = callPackage ../development/python-modules/PyLD { };
```
the packages somewhat follow alphabetic order but as you've noticed its a bit messy. Try to find a bit better place than just at the end.

apeschar commented:
Thanks again.

Fixed the stray `inherit` you pointed out, and moved the reference with a few other py* packages.

I guess I didn't really think about the point of testing my package before. Anyway, for the purpose if knowing whether the package did not break, what we have now in the way of tests is good enough.

FRidh commented:
Alright, thanks.
